cpp/src/neighbors/ivf_pq/ivf_pq_compute_similarity_impl.cuh

```diff
@@ -421,10 +432,26 @@ RAFT_KERNEL compute_similarity_kernel(uint32_t dim,
               }
               score -= q * pq_c;
             } break;
+            case distance::DistanceType::CosineExpanded: {
+              // NB: we negate the scores as we hardcoded select-topk to always compute the minimum
+              float q, c, quantized;
+              if constexpr (PrecompBaseDiff) {
+                float2 pvals = reinterpret_cast<float2*>(lut_end)[j];
+                q            = pvals.x;
+                c = pvals.y;
+              } else {
+                q = query[j];
+                c = cluster_center[j];
+              }
+              quantized = c + pq_c;
+              score -= q * quantized;
+              q_norm += q * q;
+              c_norm += quantized * quantized;
+            } break;
             default: __builtin_unreachable();
           }
         } while (++j < j_end);
-        lut_scores[i] = LutT(score);
+        lut_scores[i] = LutT(score / (sqrt(q_norm * c_norm)));
```

Comment:
Please only use `div` and `sqrt` if it is needed (i.e. only inside the `CosineExpanded` branch), because these are expensive to compute.